### PR TITLE
rename guided tours to quickstart

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Addresses ##
https://issues.redhat.com/browse/ODC-4324

## Issue ##
GuidedTours got renamed to QuickStart

## Solution ##
all components, utils, tests, actions and reducers related to GuidedTour renamed to QuickStart

## Test ##
test names altered

## Browser ##
Chrome

### CODE REVIEW

File: frontend/packages/console-app/src/components/quick-starts/DrawerComponent.tsx [hunk 1]
```diff
@@ -10,12 +10,12 @@ import {
...
-import { GuidedTourItem, GuidedTourState } from './utils/guided-tour-typings';
-import './GuidedTourDrawer.scss';
+import { QuickStartItem, QuickStartState } from './utils/quick-start-typings';
+import './QuickStartDrawer.scss';
 
 type DrawerComponentProps = {
-  guidedTour: GuidedTourItem;
-  guidedTourState?: GuidedTourState;
+  QuickStart: QuickStartItem;
+  QuickStartState?: QuickStartState;
```
Comment (sahil143) [inline frontend/packages/console-app/src/components/quick-starts/DrawerComponent.tsx:18]: props should follow camelCase naming
```suggestion
  quickStart: QuickStartItem;
  quickStartState?: QuickStartState;
```

File: frontend/packages/console-app/src/components/quick-starts/QuickStartHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+type QuickStartHeaderProps = {
+  status: string;
+  duration: number;
+  name: string;
+};
+const QuickStartHeader: React.FC<QuickStartHeaderProps> = ({ status, duration, name }) => (
+  <div className="oc-quick-start-header">
+    {name}
+    <div className="oc-quick-start-header_status">
+      {/* fix me - change badges to labels once migrated to PF4 - */}
```
Comment (sahil143) [inline frontend/packages/console-app/src/components/quick-starts/QuickStartHeader.tsx:17]: should probably do this now that PF4 PR is merged

File: frontend/packages/console-app/src/components/quick-starts/QuickStartCatalog.tsx [hunk 3]
```diff
@@ -3,47 +3,47 @@ import { Dispatch } from 'redux';
...
+  activeQSID?: string;
 };
 
 type DispatchProps = {
   onClick?: (tourId: string) => void;
 };
 
 type OwnProps = {
-  tours: GuidedTourCatalogItem[];
+  tours: QuickStartCatalogItem[];
```
Comment (nemesis09) [inline frontend/packages/console-app/src/components/quick-starts/QuickStartCatalog.tsx:26]: quickStarts: QuickStartCatalogItem[];

File: frontend/packages/console-app/src/components/quick-starts/QuickStartCatalog.tsx [hunk 4]
```diff
@@ -3,47 +3,47 @@ import { Dispatch } from 'redux';
...
         {tours.map((tour) => (
           <GalleryItem key={tour.name}>
-            <GuidedTourItem
+            <QuickStartItem
               {...tour}
-              onClick={() => onClick(!isExpanded || tour.id !== activeTourID ? tour.id : '')}
+              onClick={() => onClick(!isExpanded || tour.id !== activeQSID ? tour.id : '')}
             />
           </GalleryItem>
         ))}
```
Comment (nemesis09) [inline frontend/packages/console-app/src/components/quick-starts/QuickStartCatalog.tsx:49]: {quickStarts.map((quickStart) => (
          <GalleryItem key={quickStart.name}>
            <QuickStartItem
              {...quickStart}
              onClick={() => onClick(!isExpanded || quickStart.id !== activeQSID ? quickStart.id : '')}
            />
          </GalleryItem>
        ))}

File: frontend/packages/console-app/src/components/quick-starts/DrawerComponent.tsx [hunk 5]
```diff
@@ -24,17 +24,17 @@ type DrawerComponentProps = {
 
 const DrawerComponent: React.FC<DrawerComponentProps> = ({
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/quick-starts/DrawerComponent.tsx:25]: While we're renaming. `DrawerComponent.tsx` => `QuickStartDrawerComponent.tsx`

File: frontend/packages/console-app/src/components/quick-starts/QuickStarts.tsx [hunk 6]
```diff
@@ -1,19 +1,19 @@
 import * as React from 'react';
 import { Helmet } from 'react-helmet';
 import { PageHeading } from '@console/internal/components/utils';
-import { getGuidedToursWithStatus } from './utils/guided-tour-utils';
-import GuidedTourCatalog from './GuidedTourCatalog';
+import { getQuickStartsWithStatus } from './utils/quick-start-utils';
+import QuickStartCatalog from './QuickStartCatalog';
 
-const GuidedTours: React.FC = () => {
+const QuickStarts: React.FC = () => {
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/quick-starts/QuickStarts.tsx:7]: Rename to `QuickStartsPage.tsx`.

File: frontend/packages/console-app/src/components/quick-starts/utils/quick-start-status.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-export enum GuidedTourStatus {
+export enum QuickStartStatuses {
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/quick-starts/utils/quick-start-status.ts:1]:
```suggestion
export enum QuickStartStatus {
```

File: frontend/packages/console-app/src/components/quick-starts/utils/quick-start-typings.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  name: string;
+  duration: number;
+  description: string;
+  prerequisites?: string[];
+};
+
+export type QuickStartStatus = {
+  active?: boolean;
+  status: QuickStartStatuses;
+};
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/quick-starts/utils/quick-start-typings.ts:16]: Does this need to be a separate type. I cannot find this type being used anywhere else other than in the catalog item below. Merge it inline?

File: frontend/packages/console-app/src/redux/reducers/quick-start-reducer.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import { QuickStartSidebarActions, Actions } from '../actions/quick-start-actions';
+import { RootState } from '@console/internal/redux';
+
+export const quickStartReducerName = 'quickStart';
+
+type State = {
+  isExpanded: boolean;
+  activeQSID: string;
```
Comment (christianvogt) [inline frontend/packages/console-app/src/redux/reducers/quick-start-reducer.ts:8]:
```suggestion
  activeQuickStartID: string;
```
We aren't abbreviating to `qs` anywhere else.

or even just:
```suggestion
  activeID: string;
```

File: frontend/packages/console-app/src/redux/reducers/quick-start-reducer.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    };
+  }
+
+  return state;
+};
+
+export const isQuickStartDrawerExpanded = (state: RootState): boolean =>
+  !!state.plugins?.console?.[quickStartReducerName]?.isExpanded;
+
+export const getActiveQSID = (state: RootState): string =>
```
Comment (christianvogt) [inline frontend/packages/console-app/src/redux/reducers/quick-start-reducer.ts:30]:
```suggestion
export const getActiveQuickStartID = (state: RootState): string =>
```
We aren't abbreviating to qs anywhere else.

File: frontend/packages/console-app/src/components/quick-starts/DrawerComponent.tsx (hunk 5 above)
Comment (abhinandan13jan) [reply frontend/packages/console-app/src/components/quick-starts/DrawerComponent.tsx]: updated

File: frontend/packages/console-app/src/components/quick-starts/QuickStarts.tsx (hunk 6 above)
Comment (abhinandan13jan) [reply frontend/packages/console-app/src/components/quick-starts/QuickStarts.tsx]: updated

File: frontend/packages/console-app/src/components/quick-starts/utils/quick-start-typings.ts (hunk 8 above)
Comment (abhinandan13jan) [reply frontend/packages/console-app/src/components/quick-starts/utils/quick-start-typings.ts]: updated

File: frontend/packages/dev-console/src/components/QuickStartTile.scss [hunk 11]
```diff
@@ -1,6 +1,6 @@
 @import '../../../../public/style/vars';
 
-.odc-guidedtour-tile {
+.odc-quickstart-tile {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/QuickStartTile.scss:3]: To be consistent with the rest of the changes:

```suggestion
.odc-quick-start-tile {
```

File: frontend/packages/console-app/src/components/quick-starts/utils/quick-start-typings.ts [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+  iconURL: string;
+  altIcon?: string;
+  id: string;
+  name: string;
+  duration: number;
+  description: string;
+  prerequisites?: string[];
+};
+
+export type QuickStartStatus = {
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/quick-starts/utils/quick-start-typings.ts:11]: `QuickStartStatus` is named the same as the enum in `quick-start-status.ts`.

Can we not export this? I don't see where this is being used outside of this file.

File: frontend/public/components/masthead-toolbar.jsx [hunk 13]
```diff
@@ -268,7 +268,7 @@ class MastheadToolbarContents_ extends React.Component {
       isSection: true,
       actions: [
         {
-          component: <Link to="/tours">Guided Tours</Link>,
+          component: <Link to="/quickstart">Guided Tours</Link>,
```
Comment (rohitkrai03) [inline frontend/public/components/masthead-toolbar.jsx:271]:
```suggestion
          component: <Link to="/quickstart">Quick Starts</Link>,
```

File: frontend/packages/dev-console/src/components/QuickStartTile.tsx [hunk 14]
```diff
@@ -14,19 +14,19 @@ import {
...
+import './QuickStartTile.scss';
 
-export const HIDE_TOUR_TILE_STORAGE_KEY = 'bridge/hide-tour-tile';
+export const HIDE_QUICK_START_STORAGE_KEY = 'bridge/hide-tour-tile';
 
-const GuidedTourTile: React.FC = () => {
-  const isTourTileHidden = localStorage.getItem(HIDE_TOUR_TILE_STORAGE_KEY) === 'true';
-  const [showTile, setShowTile] = React.useState<boolean>(!isTourTileHidden);
+const QuickStartTile: React.FC = () => {
+  const isQsTileHidden = localStorage.getItem(HIDE_QUICK_START_STORAGE_KEY) === 'true';
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/QuickStartTile.tsx:23]:
```suggestion
  const isQuickStartTileHidden = localStorage.getItem(HIDE_QUICK_START_STORAGE_KEY) === 'true';
```

File: frontend/packages/dev-console/src/components/QuickStartTile.tsx [hunk 15]
```diff
@@ -14,19 +14,19 @@ import {
...
+  const [showTile, setShowTile] = React.useState<boolean>(!isQsTileHidden);
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/QuickStartTile.tsx:24]:
```suggestion
  const [showTile, setShowTile] = React.useState<boolean>(!isQuickStartTileHidden);
```

**Pull request merged**